### ISSUE TITLE
Build broken (on GNU/Linux)

### ISSUE THREAD

**Issue by jschwartzenberg** [ojdkbuild/ojdkbuild#87]
When I try to build with a simple `./gradlew` on my system, I get the following output:
```
[julius@lc-julius-rhel6 upstream_openjfx-8u]$ ./gradlew

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':buildSrc:runtimeClasspath'.
> Could not find org.antlr:antlr:3.1.3.
  Searched in the following locations:
    - file:/media/sdb1/julius/deps/openjfx/maven/org/antlr/antlr/3.1.3/antlr-3.1.3.pom
    - file:/media/sdb1/julius/deps/openjfx/maven/org/antlr/antlr/3.1.3/antlr-3.1.3.jar
  Required by:
      project :buildSrc

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
```

There is a hardcoded path to `../../deps/openjfx`. Restoring the `build.gradle` files and adding `build.properties` back allows me to do a build. It seems this was broken with commit fb9f6665102e26f0e766dc4dd36566afd25cc88b.

**Comment by ojdkbuild** [comment]
Moving the issue here from [openjfx-8](https://github.com/ojdkbuild/upstream_openjfx-8u) repo.

Builds of OpenJFX-8 in this project are done only for Windows. Build resources that are used for OpenJFX are stored in [this repo](https://github.com/ojdkbuild/deps_openjfx). Maintaining OpenJFX builds for Linux requires a non-trivial effort - it is not possible to maintain Linux builds in this project.

I know there was a somehow-working [Fedora package](https://src.fedoraproject.org/rpms/openjfx), it may be better to use it as a base for Linux builds.

Some links on OpenJFX-8: [upstream status](http://mail.openjdk.java.net/pipermail/openjfx-dev/2019-June/023352.html), [ojdkbuild 8u212+ status](https://groups.google.com/d/msg/ojdkbuild/Db-Z4Inj-ic/GhKzJGuQCgAJ).

**Comment by jschwartzenberg** [comment]
Yes, the Fedora package works. It would be nice to have a single place to keep all fixes that are backported onto OpenJFX 8. For our usecase we also might need to backport this change: http://hg.openjdk.java.net/openjfx/jfx-dev/rt/rev/e687603602c1

**Comment by ojdkbuild** [comment]
I understand the reasoning, but I think it is better to not have Linux patches in this repo. Besides being built only for Windows, it doesn't have the full history. I think a possible jfx-8 repo in javafxports account would be much better.

Thanks for the link to the CoUninitialize issue! Something similar is [included in this repo](https://github.com/ojdkbuild/upstream_openjfx-8u/commit/e020946d2cf5aa30df317119ebd52d735acd04b8) - need to check whether the patches are actually doing the same.

**Comment by jschwartzenberg** [comment]
> Thanks for the link to the CoUninitialize issue! Something similar is [included in this repo](https://github.com/ojdkbuild/upstream_openjfx-8u/commit/e020946d2cf5aa30df317119ebd52d735acd04b8) - need to check whether the patches are actually doing the same.

Nice! I was already testing our build against your OpenJDK build. I'll let you know if I still run into this issue or not.

Yes, it's indeed unfortunate that your repo doesn't have the full history. Unfortunately the javafxports copy also appears to miss some updates and yours seem to be the only one that is still being kept up to date at all. It would be very nice to have some repo that has the whole history and also contains the backported fixes that you have.

**Comment by jschwartzenberg** [comment]
I can confirm that the fix you have in place appears to be working. I could not reproduce this issue with your build.

**Comment by jschwartzenberg** [comment]
When I clone deps_openjfx and change `/../../deps/openjfx/maven/` into `/../deps_openjfx/maven/`, the build proceeds, but there is still the missing `build.properties` file. How do you handle that with your builds?

**Comment by ojdkbuild** [comment]
Thanks for the info about CoUninitialize patch.

Properties files (`build.properties` and `gradle.properties`) are pre-processed with CMake: [template](https://github.com/ojdkbuild/deps_openjfx/blob/master/resources/build.properties), [cmake directive](https://github.com/ojdkbuild/deps_openjfx/blob/master/CMakeLists.txt#L49).

**Comment by jschwartzenberg** [comment]
> I understand the reasoning, but I think it is better to not have Linux patches in this repo. Besides being built only for Windows, it doesn't have the full history.

I couldn't find an up to date mirror of the OpenJFX 8 sources, so I imported the last state from the original Mercurial repository and applied your patches and a few others on top of it. This allows me to build with WebKit enabled on RHEL 6 togeher with the devtoolset-8, python27 and rh-ruby24 software collections.

I am also looking at the Fedora package. That one is built without WebKit support and relies on gradle-local. It would be nice to have this working with WebKit, but I don't know if there is a proper way to build such an RPM with SCLs (it would be nice to do it on COPR). Doing an RPM without WebKit seems doable.

**Comment by ojdkbuild** [comment]
Sorry, but I'll close this as "wontfix".

I will continue to backport security and WebKit fixes for Windows builds in this project, but an adequate support for Linux builds is not possible - it requires quite a bit of maintenance. For OpenJFX users it may be better to switch to standalone OpenJFX11+ modules, that should have much better upstream support.

**Comment by jschwartzenberg** [comment]
Hi, that's ok. I will try to keep my repo (1) up to date with your fixes. I am not aware of any better source for OpenJFX 8. I have been doing some builds myself (2) that seem to work well for our situation. This is the standard RHEL6 version, but I removed the changes to use the system NSS, so that it uses these bits from the OpenJDK itself. On top of that there is a WebKit build from my repo. This behaves identical to your Windows builds as far as I have observed. (I tried Corretto as well, but I see some failing tests with it and they are not very actively looking at PRs.)

(1) https://github.com/jschwartzenberg/openjfx8
(2) https://schwart6.home.xs4all.nl/rhel6/openjdk/
